app/controllers/application_controller.rb

```diff
@@ -1,3 +1,25 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+  helper_method :current_user
+ 
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
+  end
+
+#  def logged_in_access_only
+#    if session[:user_id].blank?
+#      flash[:notice] = "You need to be logged in to access that feature"
+#      redirect_to root_path
+#    end
+#  end
+#  def logged_out_access_only
+#    redirect_to home_path if current_user
+#  end
+
+  def available_to(who)
```

Comment:
I get what you are doing but it is not that intuitive when you read the definition or the usages of this code. I would prefer  having (and do have, in my own code) two separate and well-named methods that handle these two cases.